packages/ckeditor5-dev-release-tools/tests/tasks/preparerepository.js

```diff
@@ -0,0 +1,321 @@
+				if ( args.length === 0 ) {
+					throw new Error( `Stub "${ stubName }" expected to receive an argument.` );
+				}
+
+				throw new Error( `No output configured for stub "${ stubName }" with the following args: ${ JSON.stringify( args ) }` );
+			}
+
+			stubs = {
+				fs: {
+					copy: sinon.stub().resolves(),
+					ensureDir: sinon.stub().resolves(),
+					writeJson: sinon.stub().resolves(),
+
+					// These stubs will reject calls without predefined arguments.
+					lstat: sinon.stub().callsFake( ( ...args ) => stubReject( 'fs.lstat', args ) ),
+					readdir: sinon.stub().callsFake( ( ...args ) => stubReject( 'fs.readdir', args ) ),
+					existsSync: sinon.stub().callsFake( ( ...args ) => stubReject( 'fs.existsSync', args ) )
+				},
+				lstat: {
+					isDir: {
+						isDirectory: sinon.stub().returns( true )
+					},
+					isNotDir: {
+						isDirectory: sinon.stub().returns( false )
+					}
+				},
+				process: {
+					cwd: sinon.stub( process, 'cwd' ).returns( 'current/working/dir' )
+				}
+			};
+
+			stubs.fs.readdir.withArgs( 'current/working/dir/release' ).resolves( [] );
+			stubs.fs.readdir.withArgs( 'current/working/dir/packages' ).resolves( packages );
+
+			mockery.enable( {
+				useCleanCache: true,
+				warnOnReplace: false,
+				warnOnUnregistered: false
+			} );
+
+			mockery.registerMock( 'fs-extra', stubs.fs );
+
+			prepareRepository = require( '../../lib/tasks/preparerepository' );
+		} );
+
+		afterEach( () => {
+			mockery.deregisterAll();
+			mockery.disable();
+			sinon.restore();
+		} );
+
+		it( 'should be a function', () => {
+			expect( prepareRepository ).to.be.a( 'function' );
+		} );
+
+		it( 'should do nothing if neither "rootPackage" or "packagesDirectory" options are defined', async () => {
+			await prepareRepository( options );
+
+			expect( stubs.fs.copy.callCount ).to.equal( 0 );
+			expect( stubs.fs.ensureDir.callCount ).to.equal( 0 );
+			expect( stubs.fs.writeJson.callCount ).to.equal( 0 );
+		} );
+
+		it( 'should ensure the existence of the output directory', async () => {
+			stubs.fs.readdir.withArgs( 'current/working/dir/packages' ).resolves( [] );
+			options.packagesDirectory = 'packages';
+
+			await prepareRepository( options );
+
+			expect( stubs.fs.ensureDir.callCount ).to.equal( 1 );
+			expect( stubs.fs.ensureDir.getCall( 0 ).args.length ).to.equal( 1 );
+			expect( stubs.fs.ensureDir.getCall( 0 ).args[ 0 ] ).to.equal( 'current/working/dir/release' );
+		} );
+
+		it( 'should throw if the output directory is not empty', async () => {
+			stubs.fs.readdir.withArgs( 'current/working/dir/release' ).resolves( [ 'someFile.txt' ] );
+			stubs.fs.readdir.withArgs( 'current/working/dir/packages' ).resolves( [] );
+			options.packagesDirectory = 'packages';
+
+			try {
+				await prepareRepository( options );
+				throw new Error( 'Expected to throw.' );
+			} catch ( err ) {
+				expect( err.message ).to.equal( 'Output directory is not empty: "current/working/dir/release"' );
+			}
```

Comment:
```suggestion
			await prepareRepository( options )
				.then(
					() => {
						throw new Error( 'Expected to throw.' );
					},
					err => {
						expect( err.message ).to.equal( 'Output directory is not empty: "current/working/dir/release"' );
					}
				);
```

I would avoid using `try/catch` and `await`. It looks bad.